### PR TITLE
Remove idx comparision in __eq__ check

### PR DESCRIPTION
**Pull request by mahaloz**

### CODE REVIEW

File: ailment/expression.py [hunk 1]
```diff
@@ -131,7 +131,6 @@ def __str__(self):
 
     def likes(self, other):
         return type(self) is type(other) and \
-            self.tmp_idx == other.tmp_idx and \
```
Comment (ltfish) [inline ailment/expression.py:134]: I don't think this is correct. Are you saying `t0 == t1`? Note that this is not the `idx` that AIL manager assigns.

File: ailment/expression.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def has_atom(self, atom, identity=True):
             return self.likes(atom)
 
     def __eq__(self, other):
-        return self.likes(other) and self.idx == other.idx
+        return self.likes(other)
```
Comment (ltfish) [inline ailment/expression.py:32]: I believe you are just making `likes == __eq__`, which negates the reason why there is a `likes()` in the first place...

File: ailment/statement.py [hunk 3]
```diff
@@ -279,7 +275,6 @@ def __init__(self, idx, target, calling_convention: Optional['SimCC']=None, prot
 
     def likes(self, other):
         return type(other) is Call and \
-               self.idx == other.idx and \
```
Comment (ltfish) [inline ailment/statement.py:282]: This is a good change.

**Comment by mahaloz** [comment]
@ltfish you are totally right, this change is bad. I'll make a new PR with the likes update. I misunderstood.

**Pull request closed without merging**